compiler/noirc_evaluator/src/ssa/plonky2_gen/div_generator.rs

```diff
@@ -76,8 +76,12 @@ impl SimpleGenerator<P2Field, 2> for VariableIntDivGenerator {
         let numerator = witness.get_target(self.numerator).to_canonical_u64();
         let denominator = witness.get_target(self.denominator).to_canonical_u64();
 
-        let quotient = numerator / denominator;
-        let remainder = numerator % denominator;
+        let mut quotient: u64 = 0;
+        let mut remainder: u64 = 0;
+        if denominator != 0 {
+            quotient = numerator / denominator;
+            remainder = numerator % denominator;
+        }
```

Comment:
How about try this:

```
        let (quotient, remainder) = if denominator != 0 {
            (numerator / denominator, numerator % denominator)
        } else {
            (u64::MAX, 0)
        };
```

... to make it similar to your other PR, where the variables are not mut.

Also, let's use u64::MAX, which is 18_446_744_073_709_551_615u64 as the result for now. This will make it less likely for someone to accidentally divide by zero and miss the fact.

(Ideally, perhaps we should figure out a way to return an error, if the code attempts to divide by zero, rather than allowing it. But let's postpone this for later.)

Reply:
I actually tried this initially, because it makes sense to approximate div by 0 with the largest number (approaching "infinity"), but it didn't work, because P2Field::from_canonical_u64 crashes with a parameter of u64:MAX. I don't know why. :(

Reply:
Oh, I see, thanks. Then please add a comment where you assign it, so that we don't wonder in the future.